tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/statesystem/TmfStateSystemAnalysisModule.java

```diff
@@ -896,4 +896,18 @@ private static boolean isCompleteTrace(ITmfTrace trace) {
         }
         return properties;
     }
+
+
+    /**
+     * @since 9.3
+     */
+    @Override
+    public void clearPersistentData() {
+        super.clearPersistentData();
+        if (fStateSystem != null) {
```

Comment:
fStateSystem is null when analysis has not been scheduled even if the state system file exists, and hence the state system is not deleted